lib/Doctrine/Migrations/Tools/TransactionHelper.php

```diff
@@ -72,11 +72,9 @@ private static function inTransaction(Connection $connection): bool
      */
     private static function getInnerConnection(Connection $connection)
     {
-        if (method_exists($connection, 'getNativeConnection')) {
-            try {
-                return $connection->getNativeConnection();
-            } catch (LogicException $e) {
-            }
+        try {
+            return $connection->getNativeConnection();
+        } catch (LogicException $e) {
         }
 
         $innermostConnection = $connection;
```

Comment:
Todo: check if the code below is still useful

Reply:
From what I understood, it will be useful until DBAL 4, when we are sure that `getNativeConnection` can no longer throw a `LogicException`.